### ISSUE TITLE
Customisable viewports

### ISSUE THREAD

**Issue by danreeves** [fffunction/backstop-crawl#6]
We currently just hardcode 
```
viewports: [
        {
            name: 'Screen',
            width: 1440,
            height: 900,
        },
    ],
```

**Comment by ataylorme** [comment]
I think this could be generic to allow for customizing the default configuration file so more than viewports can be changed

**Comment by ataylorme** [comment]
The scenario configuration is hardcoded into `lib/crawl.js` as well and I had to edit this file to be able to add `referenceUrl`, `delay` and `misMatchThreshold`. See [this commit in my fork](https://github.com/ataylorme/backstop-crawl/commit/ed0e1ca2840b46637169923784394d88a9e62d9d).

My use case is testing a staging environment with security updates applied against the production site. I'm automating this with a script that runs on cron and using Backstop JS as part of my testing. Right now I have a hardcoded `backstop.json` which I must edit to change URLs for the site. 

I'm attempting to crawl the site rather than having my `backstop.json` hardcoded.

@danreeves This tool works well but I have to tweak some hardcoded items noted above. I'd love to work with you to make things more configurable so I don't have to maintain a fork and others can easily use this tool as well. 

Perhaps a `backstop.default.json` can be detected and used if present? My idea is it would have all settings, such as `viewports`, `engine`, etc. as well as one scenario. `backstop-crawl` can then take this and copy the default scenario when crawling, populating `url` and `referenceUrl` with `referenceUrl` only being populated if two URLs are passed - e.g. `backstop-crawl https://mysite.com/ https://staging.mysite.com/`

**Comment by danreeves** [comment]
Hey @ataylorme

This sounds really good and a pr would definitely be welcome!

I like the idea of a default config to extend, perhaps it should be be `backstop.template.json` though. It'd also be nice to provide a `--template` flag that lets you specify file named differently.

I'm not sold on the second param being a URL to replace with but a named flag seems fine.

I wonder if there's a more general solution like the config file being a `.js` which exports a default config and a mapper to transform URLs ?

**Comment by ataylorme** [comment]
@danreeves I submitted a PR for what I have so far. It allows for a `backstop-crawl.js`. Basically it is an exported `backstop.json` with the addition of a `defaultScenario` key which is used to construct the scenarios in `lib/crawl.js`. Would love your thoughts on this approach.


The one I tested with looks like this:

```js
module.exports = 
{
  viewports: [
    {
      name: "phone",
      width: 320,
      height: 480
    },
    {
      name: "tablet_v",
      width: 568,
      height: 1024
    },
    {
      name: "tablet_h",
      width: 1024,
      height: 768
    },
    {
      name: "desktop",
      width: 1920,
      height: 1080
    }
  ],
  scenarios: [],
  defaultScenario: {
	label: "Default",
	url: "",
	referenceUrl: "",
	hideSelectors: [],
	selectors: [
	"document"
	],
	readyEvent: null,
	delay: 1500,
	misMatchThreshold : 0.1
},
  paths: {
    bitmaps_reference: "backstop_data/bitmaps_reference",
    bitmaps_test: "backstop_data/bitmaps_test",
    compare_data: "backstop_data/bitmaps_test/compare.json",
    casper_scripts: "backstop_data/casper_scripts"
  },
  engine: "chrome",
  report: [ "CLI" ],
  casperFlags: [],
  debug: false,
  port: 3001
}
```
> I'm not sold on the second param being a URL to replace with but a named flag seems fine.

I think allowing it as a flag instead of a second parameter makes sense and will work on that conversion next.

**Comment by ataylorme** [comment]
@danreeves the PR has been updated to allow a flag instead of a second command input. I'm stepping away for a while but would love a review and comments on the direction this is headed.

**Comment by danreeves** [comment]
Awesome, I'll take a look asap

**Comment by danreeves** [comment]
Closed by #23 🎉
